src/clj/villagebook/category/models.clj

```diff
@@ -0,0 +1,16 @@
+(ns villagebook.category.models
+  (:require [villagebook.category.db :as db]
+            [villagebook.field.db :as field-db]
+            [villagebook.organisation.models :as org-models]
+            [clojure.java.jdbc :as jdbc]
+            [villagebook.category.spec :as spec]
+            [villagebook.config :as config]))
+
+(defn create!
+  [name fields org-id user-id]
+  (jdbc/with-db-transaction [trn (config/db-spec)]
+    (if (org-models/is-owner-or-member? trn org-id user-id)
+      (let [category (db/create! trn name org-id)]
+         (field-db/create-fields! trn (:id category) fields)
```

Comment:
Maybe create a model namespace for fields as well. If later on you want to tack on some validations on fields, those would make more sense in the field.models namespace rather than here.